src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -72,7 +116,7 @@ public function loadChoiceList($value = null)
             return $this->choiceList;
         }
 
-        return $this->choiceList = new ArrayChoiceList($this->getTimezones(), $value);
+        return $this->choiceList = new ArrayChoiceList($this->getTimezones($this->regions), $value);
```

Comment:
This is an issue. The choice list is cached and reused. By doing this, you'll always get the same list when using the type multiple times in a form or nested forms.

@HeahDude could probably give you more inputs than me for about this right now, because I think it was something he had to consider when working on https://github.com/symfony/symfony/pull/18334.
You should also avoid storing options in type properties, as form type are instantiated once, it acts like a global state for the type.